sdk/personalizer/Azure.AI.Personalizer/src/Azure.AI.Personalizer.csproj

```diff
@@ -22,6 +22,8 @@
 
   <ItemGroup>
     <PackageReference Include="Azure.Core" />
```

Comment:
Tried to add temp project reference to Azure.Core, it works for DPG projects.
But when comes to mgmt projects, everyone references to Azure.ResourceManager, which also references to Azure.Core package.
I also tried to update Azure.Core as project reference in Azure.ResourceManager, then everything breaks since everyone needs Azure.Core as project reference.

My proposal is:
- Upgrade Microsoft.Bcl.AsyncInterfaces in Packages.Data.props as #44185 
- Release Azure.Core
- apply the above Azure.Core and upgrade STJ in Packages.Data.props also fix the AOT warnings
- Release Azure.Core again